### PR TITLE
Added hasOwn helper.

### PR DESCRIPTION
**Pull request by avimehta**
Branched from #6560

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -113,3 +113,15 @@ export function isEnumValue(enumObj, s) {
...
+/**
+ * Checks if the given key is a property in the map.
+ *
+ * @param {T}  obj a map like property.
+ * @param {string}  key
+ * @return {boolean}
+ * @template T
+ */
+export function hasOwn(obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key);
```
Comment (jridgewell) [inline src/types.js]: Nit: We can hoist `Object.prototype.hasOwnProperty`. https://github.com/ampproject/amphtml/pull/6592/files#diff-3a699a0719518f967e9b2df3f5549e02R18

File: src/types.js [hunk 2]
```diff
@@ -113,3 +113,15 @@ export function isEnumValue(enumObj, s) {
...
+
+/**
+ * Checks if the given key is a property in the map.
+ *
+ * @param {T}  obj a map like property.
+ * @param {string}  key
+ * @return {boolean}
+ * @template T
+ */
+export function hasOwn(obj, key) {
```
Comment (jridgewell) [inline src/types.js]: Should this be in `types.js`?

Comment (avimehta) [inline src/types.js]: `map()` is in this file and I thought this function is closely related to that. Happy to create `src/utils/object.js`.

File: src/types.js (hunk 1 above)
Comment (avimehta) [inline src/types.js]: are there any performance benefits of hoisting it out? if so, will do it, otherwise it just adds more cognitive load.

Comment (jridgewell) [inline src/types.js]: If so, they'd be mild. It's more from my underscore days, where we can't trust that the prototype won't be monkey patched by some bad actor.

File: src/types.js (hunk 2 above)
Comment (jridgewell) [inline src/types.js]: I don't really think map belongs here either. 😀

Comment (avimehta) [inline src/types.js]: ok.. in that case, I'll move both of them to a new file.

File: src/types.js (hunk 1 above)
Comment (avimehta) [inline src/types.js]: will move it out..

Comment (avimehta) [inline src/types.js]: done.

**Comment by avimehta** [comment]
PTAL and approve again? Also, I'll be OOO next two weeks, feel free to merge this if I don't merge it..

**Pull request merged**